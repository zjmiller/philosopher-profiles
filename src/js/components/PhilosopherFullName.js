import React from 'react';

function PhilosopherFullName({ philosopher }){
  return (
    <span>
    {philosopher.name.first}
    {philosopher.name.first ? ' ' : ''}
    {philosopher.name.last}
    </span>
  );
}

export default PhilosopherFullName;
