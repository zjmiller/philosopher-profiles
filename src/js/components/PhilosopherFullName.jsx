import React from 'react';
import { connect } from 'react-redux';
import getFirstNameOfPhilosopher from '../selectors/getFirstNameOfPhilosopher';
import getLastNameOfPhilosopher from '../selectors/getLastNameOfPhilosopher';

export function PhilosopherFullName({ firstName, lastName }) {
  return (
    <span>
      {firstName}
      {firstName ? ' ' : ''}
      {lastName}
    </span>
  );
}

PhilosopherFullName.propTypes = {
  firstName: React.PropTypes.string,
  lastName: React.PropTypes.string,
};

const mapStateToProps = (state, { philosopher }) => ({
  firstName: getFirstNameOfPhilosopher(state, philosopher),
  lastName: getLastNameOfPhilosopher(state, philosopher),
});

export default connect(mapStateToProps)(PhilosopherFullName);
