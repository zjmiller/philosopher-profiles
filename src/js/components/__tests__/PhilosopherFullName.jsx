/* eslint-disable no-undef */

import React from 'react';
import renderer from 'react-test-renderer';
import { PhilosopherFullName } from '../PhilosopherFullName';

test('PhilosopherFullName renders correctly', () => {
  const firstName = 'Zachary';
  const lastName = 'Miller';

  const tree = renderer.create(
    <PhilosopherFullName firstName={firstName} lastName={lastName} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
