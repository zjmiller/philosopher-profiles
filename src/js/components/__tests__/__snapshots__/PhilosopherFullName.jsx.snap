exports[`test PhilosopherFullName renders correctly 1`] = `
<span>
  Zachary
   
  Miller
</span>
`;
